src/connector/src/sink/console.rs

```diff
@@ -0,0 +1,200 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use itertools::join;
+use risingwave_common::array::{Op, StreamChunk};
+use risingwave_common::catalog::Schema;
+use risingwave_common::types::{DatumRef, ScalarRefImpl};
+
+use crate::sink::{Result, Sink};
+
+pub const CONSOLE_SINK: &str = "console";
+
+#[derive(Clone, Debug)]
+pub struct ConsoleConfig {
+    pub prefix: Option<String>,
+    pub suffix: Option<String>,
+}
+
+impl ConsoleConfig {
+    pub fn from_hashmap(values: HashMap<String, String>) -> Result<Self> {
+        Ok(ConsoleConfig {
+            prefix: values.get("prefix").cloned(),
+            suffix: values.get("suffix").cloned(),
+        })
+    }
+}
+
+#[derive(Debug)]
+pub struct ConsoleSink {
+    pub epoch: u64,
+    pub schema: Schema,
+    pub buffer: Vec<String>,
+    pub prefix: String,
+    pub suffix: String,
+}
+
+impl ConsoleSink {
+    pub fn new(config: ConsoleConfig, schema: Schema) -> Result<Self> {
+        Ok(ConsoleSink {
+            epoch: 0,
+            schema,
+            buffer: vec![],
+            prefix: config.prefix.unwrap_or_default(),
+            suffix: config.suffix.unwrap_or_default(),
+        })
+    }
+}
+
+#[async_trait]
+impl Sink for ConsoleSink {
+    async fn write_batch(&mut self, chunk: StreamChunk) -> Result<()> {
+        for (op, row_ref) in chunk.rows() {
+            let row_repr = join(row_ref.values().map(parse_datum), ",");
+            let op_repr = match op {
+                Op::Insert => "INSERT",
+                Op::UpdateDelete => "UPDATE_DELETE",
+                Op::UpdateInsert => "UPDATE_INSERT",
+                Op::Delete => "DELETE",
+            };
+            println!("{}{} [{}]{}", self.prefix, op_repr, row_repr, self.suffix);
+        }
+
+        Ok(())
+    }
+
+    async fn begin_epoch(&mut self, epoch: u64) -> Result<()> {
+        self.epoch = epoch;
+        Ok(())
+    }
+
+    async fn commit(&mut self) -> Result<()> {
+        println!(
+            "{}COMMIT [Epoch({})]{}",
+            self.prefix, self.epoch, self.suffix
+        );
+        Ok(())
+    }
+
+    async fn abort(&mut self) -> Result<()> {
+        Ok(())
+    }
+}
+
+fn parse_datum(datum: DatumRef<'_>) -> String {
+    match datum {
+        None => "NULL".to_string(),
+        Some(ScalarRefImpl::Int32(v)) => format!("Int32({})", v),
+        Some(ScalarRefImpl::Int64(v)) => format!("Int64({})", v),
```

Comment:
Is `Debug` on `ScalarRefImpl` suitable for here?

Reply:
For primitive types, the `Debug` and `Display` trait performs equally well. 

However, in case of Lists and Structs, `Display` hasn't been built in yet, and `Debug` does not provide the output we expect.

```
# formatted by ConsoleSink
INSERT [Int32(5),Utf8("Eve"),Struct(List(Utf8("36 Robinson Rd"), Utf8("160 Spear St")))]

# formatted by `Debug`
INSERT [5, "Eve", Some(List(Some(Utf8("36 Robinson Rd"))Some(Utf8("160 Spear St"))))]

```